### PR TITLE
fix(rust, python): Correct CSV row indexing

### LINKED ISSUE #5380: withRowCount() seems to reset its output
### Polars version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://crates.io/crates/polars) of Polars.

### Issue description

When you chain withRowCount to scanCSV, the row count column seems to reset after a while.
Reported from nodejs-polars. 
https://github.com/pola-rs/nodejs-polars/issues/15

### Reproducible example

see https://github.com/pola-rs/nodejs-polars/issues/15

### Expected behavior

see https://github.com/pola-rs/nodejs-polars/issues/15

### Installed versions

see https://github.com/pola-rs/nodejs-polars/issues/15

### PR DESCRIPTION
**Pull request by owrior**
- Closes #5380

### CODE REVIEW

File: py-polars/tests/unit/io/test_csv.py [hunk 1]
```diff
@@ -930,3 +930,14 @@ def test_csv_single_categorical_null() -> None:
 
     assert df.dtypes == [pl.Utf8, pl.Categorical, pl.Utf8]
     assert df.to_dict(False) == {"x": ["A"], "y": [None], "z": ["A"]}
+
+
+def test_read_csv_chunked() -> None:
```
[the remark this answers is not in the document]
Comment (ritchie46) [reply py-polars/tests/unit/io/test_csv.py:935]: Given that you create a reasonable size dataset. Can you move this test to `py-polars/tests/slow`?

**Review by ritchie46 — COMMENTED** [review]
Thanks! I have one suggestion for the test, otherwise good to go.

File: py-polars/tests/unit/io/test_csv.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (owrior) [reply py-polars/tests/unit/io/test_csv.py:935]: Ok sure thing.

[the remark this answers is not in the document]
Comment (owrior) [reply py-polars/tests/unit/io/test_csv.py]: Done!

**Pull request merged**